jest-common/src/main/java/io/searchbox/client/JestClient.java

```diff
@@ -14,7 +14,7 @@
 
     <T extends JestResult> T execute(Action<T> clientRequest) throws IOException;
 
-    <T extends JestResult> void executeAsync(Action<T> clientRequest, JestResultHandler<T> jestResultHandler);
+    <T extends JestResult> void executeAsync(Action<T> clientRequest, JestResultHandler<? super T> jestResultHandler);
```

Comment:
That way we can use more generic handlers - e.g. JestResultHandler<JestResult> when executeAsync is parametrized with DocumentResult.
By introducing new type DocumentResult I wanted to keep currently working handlers intact.